tests/test_syntax/extensions/test_tables.py

```diff
@@ -60,3 +61,855 @@ def test_empty_cells(self):
+                <tr>
+                <td><code>help()</code></td>
+                <td>Display the help window.</td>
+                </tr>
+                <tr>
+                <td><code>destroy()</code></td>
+                <td><strong>Destroy your computer!</strong></td>
+                </tr>
+                </tbody>
+                </table>
+                """
+            ),
+            extensions=['tables']
+        )
+
+    def test_align_three(self):
+        self.assertMarkdownRenders(
+            self.dedent(
+                """
+                |foo|bar|baz|
+                |:--|:-:|--:|
+                |   | Q |   |
+                |W  |   |  W|
+                """
+            ),
+            self.dedent(
+                """
+                <table>
+                <thead>
+                <tr>
+                <th style="text-align: left;">foo</th>
+                <th style="text-align: center;">bar</th>
+                <th style="text-align: right;">baz</th>
+                </tr>
+                </thead>
+                <tbody>
+                <tr>
+                <td style="text-align: left;"></td>
+                <td style="text-align: center;">Q</td>
+                <td style="text-align: right;"></td>
+                </tr>
+                <tr>
+                <td style="text-align: left;">W</td>
+                <td style="text-align: center;"></td>
+                <td style="text-align: right;">W</td>
+                </tr>
+                </tbody>
+                </table>
+                """
+            ),
+            extensions=['tables']
+        )
+
+    def test_three_columns(self):
+        self.assertMarkdownRenders(
+            self.dedent(
+                """
+                foo|bar|baz
+                ---|---|---
+                   | Q |
+                 W |   | W
+                """
+            ),
+            self.dedent(
+                """
+                <table>
+                <thead>
+                <tr>
+                <th>foo</th>
+                <th>bar</th>
+                <th>baz</th>
+                </tr>
+                </thead>
+                <tbody>
+                <tr>
+                <td></td>
+                <td>Q</td>
+                <td></td>
+                </tr>
+                <tr>
+                <td>W</td>
+                <td></td>
+                <td>W</td>
+                </tr>
+                </tbody>
+                </table>
+                """
+            ),
+            extensions=['tables']
+        )
+
+    def test_three_spaces_prefix(self):
+        self.assertMarkdownRenders(
+            """
+Three spaces in front of a table:
+
+   First Header | Second Header
+   ------------ | -------------
+   Content Cell | Content Cell
+   Content Cell | Content Cell
+
+   | First Header | Second Header |
+   | ------------ | ------------- |
+   | Content Cell | Content Cell  |
+   | Content Cell | Content Cell  |
+""",
```

Comment:
You can still wrap this in `self.dedent`, which is easier to read. The first line sets the amount of indentation to remove. As it is outdented from the table, it will work correctly.

Same goes for a number of other tests below.

Reply:
I've indented/dedented where possible. One specific line was too long in a later test and because it already contains a large number of `\`s, I opted to leave it flush for clarity instead of wrapping it with a `\` at the end of the line prior.